Homepage design enhancements

flickz opened the pull request:
Resolves https://github.com/opencollective/opencollective/issues/3321

Betree commented on components/TopBar.js:
```diff
@@ -12,8 +13,10 @@ import { Box, Flex } from './Grid';
+import { Span } from './Text';
 import TopBarProfileMenu from './TopBarProfileMenu';
 import { withUser } from './UserProvider';
+// import Container from './Container';
```
```suggestion
```

Betree commented on components/home/sections/CreateCollective.js:
```diff
@@ -0,0 +1,66 @@
+import { H4 } from '../../Text';
+
+const Wrapper = styled(Box)`
+  background-image: ${props =>
+    props.hovering
+      ? "url('/static/images/home/create-collective-bg-illustration-hover-sm.png')"
+      : "url('/static/images/home/create-collective-bg-illustration-sm.png')"};
```
That shouldn't be done this way, because it creates a flickering issue while the image first load. In `Contribute to the platform` from https://github.com/opencollective/opencollective/issues/3321, I described another way to do that that prevents this flickering issue.

Betree commented on components/home/sections/CreateCollective.js:
```diff
@@ -0,0 +1,66 @@
+          />
+        </H4>
+      </Box>
+      <Wrapper width={['288px', '283px']} height={['288px', '294']} hovering={hoverCreateCollectiveButton}>
```
```suggestion
      <Wrapper width={['288px', '283px']} height={['288px', '294px']} hovering={hoverCreateCollectiveButton}>
```

Betree commented on components/home/sections/Features.js:
```diff
@@ -64,8 +71,8 @@ const FeatureList = styled(Box)`
-  @media screen and (min-width: 64em) {
-    color: ${props => props.active && '#DC5F7D'};
+  @media screen and (min-width: 40em) {
+    /* color: ${props => props.active && '#DC5F7D'}; */
```
Can we remove the custom component if it has no custom styling?

Betree commented:
@flickz it looks like the feedback from https://github.com/opencollective/opencollective-frontend/pull/4679#pullrequestreview-458147333 > "Visual diffs" haven't been addressed, can you check?

flickz commented:
@Betree 

> New line issue
Fixed on my end, or what screen resolution are you using so I can check?

> Space between items, 160px in design VS 266px in implementation (needs to be reviewed for all the parts of this section)

<img width="1089" alt="Screenshot 2020-07-30 at 10 56 57 AM" src="https://user-images.githubusercontent.com/15707013/89269588-fc213e00-d631-11ea-880c-818f6b1b3ab8.png">

The current is `margin-top: 64px` and `margin-bottom: 64px` (=128px) I'm not sure how `266px` came in or what do you think?

 > Items order in the We are open in every way section don't match design & Font color

Both fixed

> Join us section

<img width="461" alt="Screenshot 2020-08-04 at 9 28 28 AM" src="https://user-images.githubusercontent.com/15707013/89271645-e9f4cf00-d634-11ea-864b-e6855add816e.png">

Fixed as well

Betree commented:
@flickz I'm testing through https://opencollective-frontend-git-enhancement-homepage.opencollective.now.sh. Screen resolution is 1920x1080 / using Firefox & Chrome, but I can reproduce on other resolutions. There's still a line break in the code: https://github.com/opencollective/opencollective-frontend/pull/4679/files#diff-03d163e6ae27d761d6ff700e57cd537aR88 

Also noted two misalignment with 600x700 screens:

![image](https://user-images.githubusercontent.com/1556356/89284001-5f20ce00-d64e-11ea-8cd6-e57580005f49.png)

![image](https://user-images.githubusercontent.com/1556356/89283957-4f08ee80-d64e-11ea-84f0-4033c9f672f6.png)

Betree commented:
Actually I still see all the issues mentioned above. Is there something wrong with Zeit's cache?

Betree commented:
> The current is margin-top: 64px and margin-bottom: 64px (=128px) I'm not sure how 266px came in or what do you think?

You can measure with a tool like https://chrome.google.com/webstore/detail/grid-ruler/joadogiaiabhmggdifljlpkclnpfncmj?hl=en and compare the result with what's in Figma. In design, the separation between the two images is 56px so approximately 64px. That means margin top and bottom should be 32px, not 64px.

![image](https://user-images.githubusercontent.com/1556356/89284835-a491cb00-d64f-11ea-8c52-09cb2704b8a9.png)

flickz commented:
@Betree What do you think now?

Betree commented:
- You've reduced the margin between sections to `16px` + `16px`. As per my previous comment:
> That means margin top and bottom should be 32px, not 64px.

- There's also one feedback from above that doesn't seem addressed yet:

**Design**
![image](https://user-images.githubusercontent.com/1556356/88894810-cc94bf00-d248-11ea-9b3f-f3ba341a0695.png)

**Implementation**
![image](https://user-images.githubusercontent.com/1556356/88894789-c272c080-d248-11ea-92b9-b1b3f193c519.png)

- Also there are responsiveness issues on 875x640:
![image](https://user-images.githubusercontent.com/1556356/89392870-5f7e9f00-d70a-11ea-9e58-6ae75c01056d.png)

![image](https://user-images.githubusercontent.com/1556356/89392915-673e4380-d70a-11ea-8aaa-fafd9dab475b.png)

![image](https://user-images.githubusercontent.com/1556356/89392936-6dccbb00-d70a-11ea-8cac-09c791ece57f.png)

flickz commented:
For the "Join the movement", what screen resolution are you looking at this from?

Betree commented:
@flickz The one specified above, all these responsiveness issues are for 875x640

flickz commented:
<img width="453" alt="Screenshot 2020-08-05 at 10 11 41 AM" src="https://user-images.githubusercontent.com/15707013/89394508-11669d00-d704-11ea-94de-e2f898d11850.png">

I meant for this, I've checked it many times, this is not what I'm seeing on my end.

flickz commented:
> But resolution shouldn't impact the color right? 

I'm not sure but the format does, this is `.png` might be different if I export `.svg`

Betree commented on components/home/sections/JoinUs.js:
```diff
@@ -52,99 +51,150 @@ const Wrapper = styled(Container)`
-          <SectionSubtitle color="white.full">
+        <Box width={['288px', 1, '408px']}>
+          <SectionSubtitle
+            fontSize={['20px', null, '24px']}
```
Default color for `SectionSubtitle` is `black.700`.

In the design, the font color for `Be part of the new generation of communities.` is `black.900`.
![image](https://user-images.githubusercontent.com/1556356/89395134-34e21580-d70d-11ea-87bd-ad81ebc0b587.png)

This is the difference I've described as the `Font color` diff in https://github.com/opencollective/opencollective-frontend/pull/4679#pullrequestreview-458147333

Betree replied:
The one in dev is greyish, the one in design is black. I'm only talking about the text color here, not the background image color.

flickz replied:
Okay, I understand this now. Thanks.

flickz commented:
@Betree Thanks for the modifications especially the `CreateCollective` background flicker, looks way better, I never realized that was what you had in mind.

 Memo just updated an the illustration in "What is great about Open collective", I'm adding that now. Just FYI.

Betree commented:
@flickz regarding your last commit, the arrows disappear when hovered. Is that intended?

![Peek 07-08-2020 14-22](https://user-images.githubusercontent.com/1556356/89645125-77485580-d8b9-11ea-825d-3a1174a8fb45.gif)

flickz commented:
No, I'm aware, I'll update it soon along with a new update from Memo.
